enterprise_subsidy/apps/subsidy/api.py

```diff
@@ -24,16 +28,14 @@ def create_learner_credit_subsidy(customer_uuid, unit, **kwargs):
     if kwargs.get('ledger'):
         return subsidy
 
-    from enterprise_access.apps.ledger.api import (
-        create_ledger,
-    )
     ledger = create_ledger(
         unit=unit,
         idempotency_key=create_idempotency_key_for_subsidy(subsidy),
     )
 
     subsidy.ledger = ledger
-    if kwargs.get('starting_balance'):
+    # The following condition is crafted to respect 0 as an explicit starting balance.
+    if 'starting_balance' in kwargs and kwargs['starting_balance'] is not None:
```

Comment:
This pair of functions needs overhaul anyway because starting_balance is passed as an object default to a subsidy above (but subsidies do not have a starting_balance field), and the functions do not cover all corner cases elegantly, according to my lookup table of outcomes:


```
subsidy_exists: get_or_create() gets an existing subsidy.
ledger_attached: the pre-existing subsidy had a non-null ledger FK. 
ledger_passed: a non-None ledger object was passed to the function.

subsidy_exists  ledger_attached  ledger_passed  outcome
0               n/a              0              [GOOD] no subsidy found, create subsidy with no ledger, create ledger and attach to subsidy
0               n/a              1              [GOOD] no subsidy found, create subsidy with passed ledger
1               0                0              [GOOD] subsidy found with null ledger, create ledger and attach to subsidy
1               0                1              [BAD] subsidy found with null ledger, but a ledger passed, so a subsidy with a null ledger returned
1               1                0              [BAD] subsidy found with attached ledger, but a new ledger is created anyway and the old one is orphaned
1               1                1              [WEIRD] subsidy found with attached ledger, but a potentially different ledger is passed and discarded
```